### PR TITLE
[IMP][9.0] account - Reconciliation of account move lines

### PR DESCRIPTION
**Pull request by JordiBForgeFlow**
This PR implements a migration of the reconciliations. Not tested in a multi-currency environment yet.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please squash the commits

**Comment by JordiBForgeFlow** [comment]
@pedrobaeza squashed. Errors in Travis are unrelated, and fixed in other PR's.

**Comment by JordiBForgeFlow** [comment]
@pedrobaeza This one is ready to merge.

File: addons/account/migrations/9.0.1.1/post-migration.py [hunk 1]
```diff
@@ -292,17 +292,55 @@ def account_internal_type(cr):
...
+        else:
+            move_line_ids[rec_id] += [move_line_id]
+    move_lines_2 = env['account.move.line']
+    for rec_id in move_line_ids.keys():
+        move_lines_2 = env['account.move.line'].browse(
+            [i for i in move_line_ids[rec_id]])
+        move_lines_2.auto_reconcile_lines()
+    for field in ['amount_residual', 'amount_residual_currency', 'reconciled']:
+        env.add_todo(env['account.move.line']._fields[field],
+                     move_lines_1 + move_lines_2)
```
Comment (StefanRijnhart) [inline addons/account/migrations/9.0.1.1/post-migration.py:328]: Scoping is messed up. move_lines_1 & 2 will only contain the move lines from the last reconciliation in each loop, which is not what you want. Fixed here: https://github.com/OCA/OpenUpgrade/pull/615

**Pull request merged**